release-tool.ps1

```diff
@@ -0,0 +1,456 @@
+            if ($_.Name -eq $Toolchain) {
+                $VSBaseDir = $_.InstallLocation
+                break
+            }
+        }
+    }
+    elseif ($vs.count -gt 1) {
+        # Ask the user which install to use
+        $i = 0
+        foreach ($_ in $vs) {
+            $i = $i + 1
+            $i.ToString() + ") " + $_.Name | Write-Host
+        }
+        $i = Read-Host -Prompt "Which Visual Studio installation do you want to use?"
+        $i = [Convert]::ToInt32($i, 10) - 1
+        if ($i -lt 0 -or $i -ge $vs.count) {
+            throw "Invalid selection made"
+        }
+        $VSBaseDir = $vs[$i].InstallLocation
+    }
+    
+    # Bootstrap the specified VS Toolchain
+    Import-Module "$VSBaseDir\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"
+    Enter-VsDevShell -VsInstallPath $VSBaseDir -Arch $Arch -StartInPath $Path | Write-Host
+    Write-Host # Newline after command output
+}
+
+function Invoke-Cmd([string] $command, [string[]] $options = @(), [switch] $maskargs, [switch] $quiet) {
+    $call = ('{0} {1}' -f $command, ($options -Join ' '))
+    if ($maskargs) {
+        Write-Host "$command <masked>" -ForegroundColor DarkGray
+    }
+    else {
+        Write-Host $call -ForegroundColor DarkGray
+    }
+    if ($quiet) {
+        Invoke-Expression $call > $null
+    } else {
+        Invoke-Expression $call
+    }
+    if ($LASTEXITCODE -ne 0) {
+        throw "Failed to run command: {0}" -f $command
+    }
+    Write-Host #insert newline after command output
+}
+
+function Invoke-SignFiles([string[]] $files, [string] $key, [string] $time) {
+    if (-not (Test-Path -Path "$key" -PathType leaf)) {
+        throw "Appsign key file was not found! ($key)"
+    }
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using $key"  -ForegroundColor Cyan
+    $KeyPassword = Read-Host "Key password: " -MaskInput
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' using Microsoft signtool..."
+        Invoke-Cmd "signtool" "sign -f `"$key`" -p `"$KeyPassword`" -d `"KeePassXC`" -td sha256 -fd sha256 -tr `"$time`" `"$_`"" -maskargs
+    }
+}
+
+function Invoke-GpgSignFiles([string[]] $files, [string] $key) {
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using GPG key $key" -ForegroundColor Cyan
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' and creating DIGEST..."
+        Remove-Item "$_.sig"
+        Invoke-Cmd "gpg" "--output `"$_.sig`" --armor --local-user `"$key`" --detach-sig `"$_`""
+        $FileName = (Get-Item $_).Name
+        (Get-FileHash "$_" SHA256).Hash + " *$FileName" | Out-File "$_.DIGEST" -NoNewline
+    }
+}
+
+
+# Handle errors and restore state
+$OrigDir = $(Get-Location).Path
+$OrigBranch = $(git rev-parse --abbrev-ref HEAD)
+$ErrorActionPreference = 'Stop'
+trap {
+    Write-Host "Restoring state..." -ForegroundColor Yellow
+    $(git checkout $OrigBranch)
```

Comment:
See subshell syntax remark earlier.
You should also make sure that a trap never throws errors.

Reply:
$( ) is referred to as "Sub Expression"
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators?view=powershell-7.2#subexpression-operator--

& is "Call Operator"
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators?view=powershell-7.2#call-operator-

I converted everything to & or Invoke-Command